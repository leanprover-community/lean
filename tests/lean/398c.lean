class ring (α) extends has_one α

variables (α : Type) [ring α]
set_option trace.class_instances true
-- this should only synthesize `has_one α` once,
-- the second time it should already be cached
#check (1 : α) = 1