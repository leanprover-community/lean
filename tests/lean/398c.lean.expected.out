[class_instances]  class-instance resolution trace
[class_instances] (0) ?x_0 : has_one α := @ring.to_has_one ?x_1 ?x_2
[class_instances] (1) ?x_2 : ring α := _inst_1
[class_instances] caching instance for ring α
_inst_1
[class_instances] caching instance for has_one α
@ring.to_has_one α _inst_1
[class_instances] cached instance for has_one α
@ring.to_has_one α _inst_1
1 = 1 : Prop
